src/components/Button/BistateButton.tsx

```diff
@@ -0,0 +1,76 @@
+/*! Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license. */
+import '../../yamui';
+import * as React from 'react';
+import Button, { ButtonProps, FabricButtonType } from './Button';
+
+export type CustomizableProps = 'color' | 'text' | 'ariaLabel' | 'icon' | 'iconPosition' | 'className';
+
+export type CustomizableButtonProps = Pick<ButtonProps, CustomizableProps>;
+
+export interface BistateButtonOwnProps {
+  activeProps: CustomizableButtonProps;
+}
+
+export type BistateButtonProps = BistateButtonOwnProps & ButtonProps;
+
+export interface BistateButtonState {
+  hasFocus: boolean;
+  isHovering: boolean;
+}
+
+const inactiveState = { hasFocus: false, isHovering: false };
+
+/**
+ * A `Button` that renders differently when 'active'; if the mouse is hovering or it has focus
+ */
+export default class BistateButton extends React.Component<BistateButtonProps, BistateButtonState> {
+  constructor(props: BistateButtonProps) {
+    super(props);
+    this.state = { ...inactiveState };
+  }
+
+  public render() {
+    const propsForState = this.isActive() ? this.props.activeProps : undefined;
+    const { children, activeProps, ...buttonProps } = this.props;
+    const props: ButtonProps = {
+      fullWidth: true,
+      ...buttonProps,
+      ...propsForState,
+      onFocus: this.onFocus,
+      onMouseEnter: this.onMouseEnter,
+      onBlur: this.onBlur,
+      onMouseLeave: this.onMouseLeave,
+    };
+    return <Button {...props} />;
+  }
+
+  private isActive = () => this.state.isHovering || this.state.hasFocus;
+
+  private onFocus = (e: React.FocusEvent<FabricButtonType>) => {
+    if (this.props.onFocus) {
+      this.props.onFocus(e);
+    }
+    this.setState({ hasFocus: true });
```

Comment:
i think these handlers should `setState` before the callback, giving the user immediate feedback in case somebody does something stupid in a callback which would cause a delay. probably not a big deal but i'd switch it

Reply:
I dont think state reconciliation will happen until this event handler has finished so not sure it matters. Dont remember react internals enough to be sure